app/modules/dlc/services/RemoteFileService.scala

```diff
@@ -0,0 +1,38 @@
+package modules.dlc.services
+
+import javax.inject.Inject
+
+import modules.dlc.models.RemoteFile
+import play.api.Logger
+import play.api.libs.ws.WSClient
+
+import scala.concurrent.{ExecutionContext, Future}
+
+trait RemoteFileService {
+  def checkAvailability(file: RemoteFile): Future[Boolean]
+}
+
+class RemoteFileServiceFactory @Inject() (wsClient : WSClient)(implicit ec: ExecutionContext) {
+
+  private val logger = Logger(classOf[RemoteFileServiceFactory])
+
+  def checkAvailability(file: RemoteFile): Future[Boolean] = {
+    val fileService =  mapping.find { case(key,value) => file.url.startsWith(key) }.map(_._2)
+
+    fileService.map { service =>
+      service.checkAvailability(file)
+    } getOrElse {
+      logger.error(s"unsupported file hoster for url: ${file.url}")
```

Comment:
When there is no plugin, then we could also check the remote url for == 200
better than nothing.